controllers/argocd/notifications/deployment.go

```diff
@@ -0,0 +1,212 @@
+
+	name := argoutil.GenerateUniqueResourceName(nr.Instance.Name, nr.Instance.Namespace, ArgoCDNotificationsControllerComponent)
+
+	notificationEnv := nr.Instance.Spec.Notifications.Env
+	notificationEnv = argoutil.EnvMerge(notificationEnv, ProxyEnvVars(), false)
+
+	podSpec := corev1.PodSpec{
+		Volumes: []corev1.Volume{
+			{
+				Name: TLSCerts,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: common.ArgoCDTLSCertsConfigMapName,
+						},
+					},
+				},
+			},
+			{
+				Name: ArgoCDRepoServerTLS,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: common.ArgoCDRepoServerTLSSecretName,
+						Optional:   argoutil.BoolPtr(true),
+					},
+				},
+			},
+		},
+		Containers: []corev1.Container{{
+			Command:         GetNotificationsCommand(nr.Instance),
+			Image:           GetArgoContainerImage(nr.Instance),
+			ImagePullPolicy: corev1.PullAlways,
+			Name:            common.ArgoCDNotificationsControllerComponent,
+			Env:             notificationEnv,
+			Resources:       GetNotificationsResources(nr.Instance),
+			LivenessProbe: &corev1.Probe{
+				ProbeHandler: corev1.ProbeHandler{
+					TCPSocket: &corev1.TCPSocketAction{
+						Port: intstr.IntOrString{
+							IntVal: int32(9001),
+						},
+					},
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				AllowPrivilegeEscalation: argoutil.BoolPtr(false),
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{
+						CapabilityDropAll,
+					},
+				},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      TLSCerts,
+					MountPath: VolumeMountPathTLS,
+				},
+				{
+					Name:      ArgoCDRepoServerTLS,
+					MountPath: VolumeMountPathRepoServerTLS,
+				},
+			},
+			WorkingDir: WorkingDirApp,
+		}},
+		SecurityContext: &corev1.PodSecurityContext{
+			RunAsNonRoot: argoutil.BoolPtr(true),
+		},
+	}
+
+	AddSeccompProfileForOpenShift(nr.Logger, nr.Client, &podSpec)
+
+	deploymentRequest := workloads.DeploymentRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: nr.Instance.Namespace,
+			Labels: map[string]string{
+				common.AppK8sKeyName: name,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				Spec: podSpec,
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						common.AppK8sKeyName: name,
+					},
+				},
+			},
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					common.AppK8sKeyName: name,
+				},
+			},
+		},
+
+		Client:    nr.Client,
+		Mutations: []mutation.MutateFunc{mutation.ApplyReconcilerMutation},
+	}
+
+	if replicas := GetArgoCDNotificationsControllerReplicas(nr); replicas != nil {
+		deploymentRequest.Spec.Replicas = replicas
+	}
```

Comment:
@ciiay lets try and avoid making changes to the deployment received from the workloads package, as this runs the risk of overwriting any mutations that have been executed already (mutations are meant to be run at the very last minute to ensure they are not overwritten)

I would suggest setting the replicas while putting together the `deploymentRequest`